sopel/plugins/rules.py

```diff
@@ -1250,39 +1250,8 @@ def has_alias(self, name):
         return name in self._aliases
 
     def escape_name(self, name):
-        """Escape the provided name if needed.
-
-        .. note::
-
-            Until now, Sopel has allowed command name to be regex pattern.
-            It was mentioned in the documentation without much details, and
-            there were no tests for it.
-
-            In order to ensure backward compatibility with previous versions of
-            Sopel, we make sure to escape command name only when it's needed.
-
-            **It is not recommended to use a regex pattern for your command
-            name. This feature will be removed in Sopel 8.0.**
-
-        """
-        if set('.^$*+?{}[]\\|()') & set(name):
-            # the name contains a regex pattern special character
-            # we assume the user knows what they are doing
-            try:
-                # make sure it compiles properly
-                # (nobody knows what they are doing)
-                re.compile(name)
-            except re.error as error:
-                original_name = name
-                name = re.escape(name)
-                LOGGER.warning(
-                    'Command name "%s" is an invalid regular expression '
-                    'and will be replaced by "%s": %s',
-                    original_name, name, error)
-        else:
-            name = re.escape(name)
-
-        return name
+        """Escape the provided name."""
+        return re.escape(name)
```

Comment:
Yeah, I don't know. It made sense to have that interface before. I wonder if we'll ever need it in the future. Right now it doesn't feel like it, but I've this weird intuition that we might?

I'm conflicted, and I'm working with gut feeling here. Not sure that's enough.

Your call!

Reply:
Well… It existed to maybe-escape the name when regex was still allowed. Now, there's no need for any of that logic. Regex command names are explicitly not supported any more, and it's silly to have a wrapper function. Remember the pain of `web.get()` and friends?

Since this is internal stuff, I think it's fine to just inline what the helper is doing. If we need it back at some point for more complicated logic again, it's not that big of a deal.